SignalServiceKit/src/Messages/TSMessagesManager.m

```diff
@@ -494,6 +495,14 @@ - (void)handleEnvelope:(OWSSignalServiceProtosEnvelope *)envelope plaintextData:
     if (envelope.hasContent) {
         OWSSignalServiceProtosContent *content = [OWSSignalServiceProtosContent parseFromData:plaintextData];
         DDLogInfo(@"%@ handling content: <Content: %@>", self.tag, [self descriptionForContent:content]);
+        
+        if ([content hasProfileKey]) {
+            NSData *profileKey = [content profileKey];
+            NSString *recipientId = envelope.source;
+            [OWSProfilesManager.sharedManager setProfileKey:profileKey
```

Comment:
Yeah, I've been thinking that and over the course of the PRs I've been consolidating around "profile state can be read on any thread but can only be written on the main thread".  But I've been tempted to move all writes to a serial queue.  Reads could still happen synchronously on any thread.  I haven't done it yet because I didn't want to prematurely optimize.  I'll add a card so we follow up.